pkg/operator/controllers/pullsecret/pullsecret_controller.go

```diff
@@ -189,3 +178,143 @@ func (r *PullSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+			delete(secret.Data, corev1.DockerConfigJsonKey)
+		}
+	}
+
+	fixedData, update, err := pullsecret.Merge(string(secret.Data[corev1.DockerConfigJsonKey]), string(operatorSecret.Data[corev1.DockerConfigJsonKey]))
+	if err != nil {
+		return nil, NoAction, err
+	}
+
+	if !create && !update {
+		return secret, NoAction, nil
+	}
+
+	if !create && update {
+		action = UpdatePullSecret
+	}
+
+	secret.Data[corev1.DockerConfigJsonKey] = []byte(fixedData)
+	return secret, action, nil
+}
+
+// emitGlobalPullSecretChange performs update of the secret when the user pull secret is broken or require recreation
+func (r *PullSecretReconciler) emitGlobalPullSecretChange(ctx context.Context, secret *corev1.Secret, action PullSecretAction) error {
+	if action == NoAction {
+		// no action required
+		return nil
+	}
+
+	if action == RecreatePullSecret {
+		// unfortunately the type field is immutable.
+		err := r.kubernetescli.CoreV1().Secrets(secret.Namespace).Delete(ctx, secret.Name, metav1.DeleteOptions{})
+		if err != nil && !kerrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	if action == CreatePullSecret || action == RecreatePullSecret {
+		r.log.Info("re-creating pull secret")
+		_, err := r.kubernetescli.CoreV1().Secrets(secret.Namespace).Create(ctx, secret, metav1.CreateOptions{})
+
+		return err
+	}
+
+	r.log.Info("updating pull secret")
+	_, err := r.kubernetescli.CoreV1().Secrets(secret.Namespace).Update(ctx, secret, metav1.UpdateOptions{})
+	return err
+}
+
+func (r *PullSecretReconciler) updateRedHatKeyCondition(secret *corev1.Secret) (*status.Condition, error) {
+	// parse keys and validate JSON
+	parsedKeys, err := pullsecret.UnmarshalSecretData(secret)
+	foundKey := false
+	failed := false
+	if err != nil {
+		r.log.Info("pull secret is not valid json - recreating")
+		failed = true
+	} else {
+		foundKey = r.checkRHRegistryKey(parsedKeys)
+	}
+
+	keyCondition := r.keyCondition(failed, foundKey)
+	return keyCondition, nil
+}
+
+// checkRHRegistryKey checks whether the rhRegistry key:
+//   - redhat.registry.io
+// is present in the pullSecret
+func (r *PullSecretReconciler) checkRHRegistryKey(psData *pullsecret.SerializedAuthMap) (foundKey bool) {
+	rhKey := "registry.redhat.io"
+
+	if psData == nil {
+		return foundKey
+	}
+
+	if auth, ok := psData.Auths[rhKey]; ok && len(auth.Auth) > 0 {
+		foundKey = true
+	}
+
+	return foundKey
+}
+
+func (r *PullSecretReconciler) keyCondition(failed bool, foundKey bool) *status.Condition {
+	keyCondition := &status.Condition{
+		Type:   arov1alpha1.RedHatKeyPresent,
+		Status: corev1.ConditionFalse,
+		Reason: "CheckFailed",
+	}
+
+	if failed {
+		keyCondition.Message = "Cannot parse pull-secret"
+		return keyCondition
+	}
+
+	keyCondition.Reason = "CheckDone"
+
+	if !foundKey {
+		keyCondition.Message = "No Red Hat key found in pull-secret"
+		return keyCondition
+	}
+
+	keyCondition.Status = corev1.ConditionTrue
+	keyCondition.Message = "Red Hat registry key present in pull-secret"
```

Comment:
This should give reader enough information to understand secrets content. And it should be machine readable (`,` separate values or smth like this)